pallets/ds-maps/src/lib.rs

```diff
@@ -68,20 +68,48 @@ impl<
         self.north_west.get_distance_vector(self.south_east)
     }
     
-    // Check tests for fn explanation
-    pub fn zone_intersects(self, target: Rect2D<Coord>) -> bool {
-        !(self.south_east.lon < target.north_west.lon || 
-          self.north_west.lon > target.south_east.lon ||
-          self.south_east.lat < target.north_west.lat || 
-          self.north_west.lat > target.south_east.lat)
+    /// True if this rect intersects other, excluding edges.
+    pub fn intersects_rect(self, target: Rect2D<Coord>) -> bool {
+        !(self.south_east.lon <= target.north_west.lon || 
+          self.north_west.lon >= target.south_east.lon ||
+          self.south_east.lat <= target.north_west.lat || 
+          self.north_west.lat >= target.south_east.lat)
+    }
+
+    /// True, if given point lies inside the rect, excluding edges. 
+    pub fn is_point_inside(&self, target: Point2D<Coord>) -> bool {
+        !(self.south_east.lon <= target.lon || 
+          self.north_west.lon >= target.lon ||
+          self.south_east.lat <= target.lat || 
+          self.north_west.lat >= target.lat)
+    }
+}
+
+#[cfg(test)]
+mod rect_tests {
+    use super::*;
+    use crate::tests::{construct_custom_rect, coord};
+    #[test]
+    fn rect_intersection_is_correct() {
```

Comment:
Worth addin separate tests for "rect A fully inside B" + "rect B fully inside A".

Sometimes it is better to choose corrdinates so that X != Y (below here `(0,0)-(2,2)` is bad in this respect) to be able to properly test that you didn't mix up lat/lon somewhere inside function.